va/va_vpp.h

```diff
@@ -1447,6 +1449,103 @@ typedef struct _VAProcFilterCap3DLUT {
+     * or similar usage scenario.
+     */
+    VAProcSuperResolutionSharpen,
+    /**
+     * \brief Super Resolution Artifact Removal Mode.
+     * In this mode, Super Resolution is optimized or trained to remove encoding artifacts
+     * (usually limited network bandwidth) with medium sharpness level.
+     * This mode is recommended to be used in video surveillance simliar usage secenario
+     * which may have camera noise and encoding artifacts in the video stream
+     * due to limited network bandwidth.
+     */
+    VAProcSuperResolutionArtifactRemoval,
+} VAProcSuperResolutionMode;
+
+/**
+ *  \brief Super Resolution(SR) filter parametrization.
+ */
+typedef struct _VAProcFilterParameterBufferSuperResolution {
+    /**
+     * \brief Filter type. Shall be set to #VAProcFilterSuperResolution.
+     */
+    VAProcFilterType                 type;
+    /**
+     * \brief Super Resolution mode, should be one of VAProcSuperResolutionXXX
+     */
+    VAProcSuperResolutionMode        mode;
+    /**
+     *\brief reserved bytes for future use, must be zero
+     */
+    uint32_t                         va_reserved[VA_PADDING_HIGH];
+} VAProcFilterParameterBufferSuperResolution;
+
+/**
+ * \brief Capabilities specification for the Super Resolution filter.
+ *
+ * Super resolution filter capabilities can be checked with vaQueryVideoProcFilterCaps.
+ *
+ * \pseudo code Query the number of Caps first, then Query Caps
+ * unsigned int num_sr_caps = 0;
+ * VAProcFilterCapSuperResolution* sr_caps = nullptr;
+ * vaQueryVideoProcFilterCaps(va_dpy, vpp_ctx,
+ *     VAProcFilterSuperResolution,
+ *     &sr_caps, &num_sr_caps);
+ * sr_caps = new vaQueryVideoProcFilterCaps[num_sr_caps];
+ * vaQueryVideoProcFilterCaps(va_dpy, vpp_ctx,
+ *     VAProcFilterSuperResolution,
+ *     &sr_caps, &num_sr_caps);
+ */
+typedef struct _VAProcFilterCapSuperResolution {
+    /** \brief Super Resolution mode, should be one of VAProcSuperResolutionXXX.*/
+    VAProcSuperResolutionMode            mode;
+    /** \brief The maximum value of input width SR filter is capable of supporting.*/
+    uint32_t                             max_input_width;
+    /** \brief The maximum value of input height SR filter is capable of supporting.*/
+    uint32_t                             max_input_height;
+    /**
+     * \brief The maximum value of scaling ratio SR filter is capable of supporting.
+     *
+     * If output resolution / input resolution > max_scaling_ratio, it is up to
+     * backend implementation how to handle this case.
+     * Here, lists two possible options for the reference.
+     * Option #1: return VA_STATUS_ERROR_RESOLUTION_NOT_SUPPORTED;
+     * Option #2: input resolution * max_scaling_ratio with SR, the remaining
+     * scaling ratio could be performed by other scaling method if the backend
+     * can ganrantee the scaling quality.
+     */
+    float                                max_scaling_ratio;
```

Comment:
Because this is the capability returned from the backend driver, so `resolution / input resolution > max_scaling_ratio` shouldn't be allowed.

Reply:
max_scaling_ratio means the maximum scaling ratio which SR can supports.